tests/test_state.py

```diff
@@ -39,49 +38,12 @@ def get_dummy_state(model: ComposerModel, train_dataloader: DataLoader):
     return state
 
 
-def _check_item(item1: Any, item2: Any, path: str):
-    if item1 is None:
-        assert item2 is None, f"{path} differs: {item1} != {item2}"
-        return
-    if isinstance(item1, (str, float, int, bool)):
-        assert type(item1) == type(item2)
-        assert item1 == item2, f"{path} differs: {item1} != {item2}"
-        return
-    if isinstance(item1, torch.Tensor):
-        assert isinstance(item2, torch.Tensor)
-        # Using a high tolerance, as deepspeed non-determinisim can cause
-        # metric values to be off.
-        assert item1.allclose(item2, rtol=0.1, atol=0.1), f"{path} differs"
-        return
-    if isinstance(item1, dict):
-        assert isinstance(item2, dict), f"{path} differs: {item1} != {item2}"
-        _check_dict_recursively(item1, item2, path)
-        return
-    if isinstance(item1, list):
-        assert isinstance(item2, list), f"{path} differs: {item1} != {item2}"
-        _check_list_recursively(item1, item2, path)
-        return
-    raise NotImplementedError(f"Unsupported item type: {type(item1)}")
-
-
-def _check_list_recursively(list1: List[Any], list2: List[Any], path: str):
-    assert len(list1) == len(list2), f"{path} differs: {list1} != {list2}"
-    for i, (item1, item2) in enumerate(zip(list1, list2)):
-        _check_item(item1, item2, f"{path}/{i}")
-
-
-def _check_dict_recursively(dict1: Dict[str, Any], dict2: Dict[str, Any], path: str):
-    assert len(dict1) == len(dict2), f"{path} differs: {dict1} != {dict2}"
-    for k, val1 in dict1.items():
-        val2 = dict2[k]
-        _check_item(val1, val2, f"{path}/{k}")
-
-
 def assert_state_equivalent(state1: State, state2: State):
     assert state1.serialized_attributes == state2.serialized_attributes
     assert state1.is_model_deepspeed == state2.is_model_deepspeed
 
-    _check_dict_recursively(state1.state_dict(), state2.state_dict(), "")
+    # Using a loose tolerance as GPU determinism does not work properly
```

Comment:
Surprised a bit about the need here, where is that coming from?

Reply:
On the resent and gpt2 checkpointing tests, I was unable to get an exact comparison to pass.